src/test/java/uk/gov/justice/digital/delius/controller/secure/AttendanceResourceAPITest.java

```diff
@@ -0,0 +1,119 @@
+import uk.gov.justice.digital.delius.service.AttendanceService;
+import uk.gov.justice.digital.delius.service.AttendanceServiceTest;
+
+@SuppressWarnings("OptionalGetWithoutIsPresent")
+@RunWith(MockitoJUnitRunner.class)
+public class AttendanceResourceAPITest {
+
+    private static final Long SOME_EVENT_ID = 12342L;
+    private static final Long SOME_CONTACT_ID_1 = 7856L;
+    private static final Long SOME_CONTACT_ID_2 = 8756L;
+
+    @Mock
+    private AttendanceService attendanceService;
+
+    @Before
+    public void setup() {
+        RestAssuredMockMvc.standaloneSetup(
+                new AttendanceResource(attendanceService),
+                new SecureControllerAdvice()
+        );
+    }
+
+    @Test
+    public void givenNoMatchingEventThenRespondWithStatusNotFound() {
+        final Long nonMatchingEventId = 99L;
+        final String expectedMsg = String.format(AttendanceResource.MSG_ATTENDANCES_NOT_FOUND, nonMatchingEventId);
+        when(attendanceService.getContactsForEvent(eq(nonMatchingEventId), any(LocalDate.class))).thenReturn(Optional.empty());
+
+        given()
+            .when()
+            .get("/secure/contacts/" + nonMatchingEventId + "/attendances/")
+            .then()
+            .log().ifValidationFails()
+            .statusCode(HttpStatus.NOT_FOUND.value())
+            .body("developerMessage", containsString(expectedMsg));
+
+        verify(attendanceService).getContactsForEvent(eq(nonMatchingEventId), any(LocalDate.class));
+        verifyNoMoreInteractions(attendanceService);
+    }
+
+    @Test
+    public void getMultipleAttendancesReturnsOk() {
+
+        final LocalDate attendanceDate1 = LocalDate.of(2019, Month.AUGUST, 24);
+        final LocalDate attendanceDate2 = LocalDate.of(2020, Month.FEBRUARY, 29);
+        final Contact contact1 = AttendanceServiceTest.getContactEntity(SOME_CONTACT_ID_1, attendanceDate1, null, null);
+        final Contact contact2 = AttendanceServiceTest.getContactEntity(SOME_CONTACT_ID_2, attendanceDate2, "1", "1");
+        final List<Contact> contacts = Arrays.asList(contact1, contact2);
+
+        when(attendanceService.getContactsForEvent(eq(SOME_EVENT_ID), any(LocalDate.class)))
+            .thenReturn(Optional.of(contacts));
+
+        // Act
+        final Attendances actual = given()
+            .contentType(APPLICATION_JSON_VALUE)
+            .when()
+            .get(String.format("/secure/contacts/%s/attendances/", SOME_EVENT_ID))
+            .then()
+            .statusCode(HttpStatus.OK.value())
+            .extract()
+            .body()
+            .as(Attendances.class);
+
+        // assert
+        assertThat(actual.getAttendances().stream()).isNotEmpty().hasSize(2);
+        assertThat(actual.getAttendances().stream()).doesNotContainNull().doesNotHaveDuplicates();
+        final Attendance attendance1 = actual.getAttendances().stream().filter(att -> SOME_CONTACT_ID_1.equals(att.getContactId())).findFirst().get();
```

Comment:
very minor but I would replace `.get()` with `.orElseThrow()` (my guess is you have small Intellij warning)

Reply:
Yes - but in a test, the test would fail in any case.

Reply:
It's test so won't happen unless test is failing in which case the test will fail.